adapters/Unity/UnityAdapter/GADMAdapterUnityBannerAd.m

```diff
@@ -49,27 +52,36 @@ - (nonnull instancetype)initWithGADMAdNetworkConnector:(nonnull id<GADMAdNetwork
 - (void)loadBannerWithSize:(GADAdSize)adSize {
   id<GADMAdNetworkConnector> strongConnector = _connector;
   id<GADMAdNetworkAdapter> strongAdapter = _adapter;
-
-  if (!strongConnector || !strongAdapter) {
-    NSLog(@"Adapter Error: No GADMAdNetworkConnector nor GADMAdNetworkAdapter found.");
+  
+  if (!strongConnector) {
+    NSLog(@"Unity Ads Adapter Error: No GADMAdNetworkConnector found.");
+    return;
+  }
+  
+  if (!strongAdapter) {
+    NSLog(@"Unity Ads Adapter Error: No GADMAdNetworkAdapter found.");
+    return;
+  }
+    
+  _gameID = [[[strongConnector credentials] objectForKey:kGADMAdapterUnityGameID] copy];
+  _placementID = [strongConnector.credentials[kGADMAdapterUnityPlacementID] copy];
+  if (!_gameID || !_placementID) {
+    NSError *error = GADMAdapterUnityErrorWithCodeAndDescription(
+           GADMAdapterUnityErrorInvalidServerParameters, @"Game ID and Placement ID cannot be nil.");
+    [strongConnector adapter:strongAdapter didFailAd:error];
     return;
   }
 
   if (![UnityAds isInitialized]) {
-    NSString *gameID = [strongConnector.credentials[kGADMAdapterUnityGameID] copy];
-    [GADMAdapterUnitySingleton.sharedInstance initializeWithGameID:gameID];
+    [[GADMAdapterUnity alloc] initializeWithGameID:_gameID withInitDelegate:Nil];
```

Comment:
I would assume that the banner ad will fail to load or initialize if the UnityAds fail to initialize here?

Reply:
Yes, if UnityAds failed to initialize banner load call will fail.